src/core/diags/factory_diags.cpp

```diff
@@ -600,10 +677,54 @@ void Diags::AlarmFired(void)
     }
 }
 
+void Diags::OutputReceivedFrame(const otRadioFrame *aFrame)
+{
+    VerifyOrExit(mReceiveConfig.mIsEnabled && (aFrame != nullptr));
+
+    Output("%u", mReceiveConfig.mReceiveCount++);
+
+    if (mReceiveConfig.mShowRssi)
+    {
+        Output(", rssi:%d", aFrame->mInfo.mRxInfo.mRssi);
+    }
+
+    if (mReceiveConfig.mShowLqi)
+    {
+        Output(", lqi:%u", aFrame->mInfo.mRxInfo.mLqi);
+    }
+
+    if (mReceiveConfig.mShowPsdu)
+    {
+        static constexpr uint16_t kBufSize = 255;
+        char                      buf[kBufSize];
+        StringWriter              writer(buf, sizeof(buf));
+
+        writer.AppendHexBytes(aFrame->mPsdu, aFrame->mLength);
+        Output(", len:%u, psdu:%s", aFrame->mLength, buf);
```

Comment:
Outputting PSDU as hex bytes is fine.

We can also consider using `GenerateNextHexDumpLine` helper methods. 

https://github.com/openthread/openthread/blob/c2d5265c6834544d07d0f5624c9d6d0b99ec48b6/src/core/common/log.hpp#L376-L401

Reply:
I didn't use the `GenerateNextHexDumpLine()` for two reasons:
1. The output format of the `GenerateNextHexDumpLine()`  is not easy for the script to decode.
2. When the radio receives lots of frames, displaying one frame per line will make the output clearer.